velox/functions/sparksql/String.h

```diff
@@ -1556,4 +1556,44 @@ struct Empty2NullFunction {
   }
 };
 
+/// luhn_check(input) -> boolean
+///
+/// Checks that a string of digits is valid according to the Luhn algorithm.
+template <typename T>
+struct LuhnCheckFunction {
```

Comment:
Would you please extract this function to a separate file as the String.h is large?

Reply:
done